daphne/src/roles.rs

```diff
@@ -263,6 +269,11 @@ pub trait DapLeader<S>: DapAuthorizedSender<S> + DapAggregator<S> {
             return Err(DapAbort::BatchOverlap);
         }
 
+        // Check whether the DAP version in the request matches the one in task config.
+        if task_config.version != req.version || req.version == DapVersion::Unknown {
+            return Err(DapAbort::InvalidProtocolVersion);
+        }
```

Comment:
In all of these methods, the version check needs to be the very first thing that happens. Why?

Reply:
This is because (1) the version must not be "Unknown" to be able to decode the data correctly and (2) depending on the version, the struct the data is decoding to may be different, right?

Reply:
Yes.